consensus/misc/eip1559/eip1559.go

```diff
@@ -47,48 +49,53 @@ func VerifyEip1559Header(config *params.ChainConfig, parent, header *types.Heade
-	return big.NewInt(0)
+	blockNumber := new(big.Int).Add(parent.Number, common.Big1)
 
-	/*
-		We temporarily set the base fee to 0, the following logic may be enabled in
-		the future hardfork.
+	if !config.IsShanghai(blockNumber) {
+		return big.NewInt(0)
+	}
 
-		// If the current block is the first EIP-1559 block, return the InitialBaseFee.
-		if !config.IsLondon(parent.Number) {
-			return new(big.Int).SetUint64(params.InitialBaseFee)
-		}
+	// Current block is the first block in Shanghai, return initial base fee
+	if !config.IsShanghai(parent.Number) {
+		return new(big.Int).SetUint64(params.InitialBaseFee)
+	}
 
-		var (
-			parentGasTarget          = parent.GasLimit / params.ElasticityMultiplier
-			parentGasTargetBig       = new(big.Int).SetUint64(parentGasTarget)
-			baseFeeChangeDenominator = new(big.Int).SetUint64(params.BaseFeeChangeDenominator)
+	var (
+		parentGasTarget          = parent.GasLimit / params.ElasticityMultiplier
+		parentGasTargetBig       = new(big.Int).SetUint64(parentGasTarget)
+		baseFeeChangeDenominator = new(big.Int).SetUint64(params.BaseFeeChangeDenominator)
+	)
+	// If the parent gasUsed is the same as the target, the baseFee remains unchanged.
+	if parent.GasUsed == parentGasTarget {
+		return new(big.Int).Set(parent.BaseFee)
+	}
+	if parent.GasUsed > parentGasTarget {
+		// If the parent block used more gas than its target, the baseFee should increase.
+		gasUsedDelta := new(big.Int).SetUint64(parent.GasUsed - parentGasTarget)
+		x := new(big.Int).Mul(parent.BaseFee, gasUsedDelta)
+		y := x.Div(x, parentGasTargetBig)
+		baseFeeDelta := math.BigMax(
+			x.Div(y, baseFeeChangeDenominator),
+			common.Big1,
 		)
-		// If the parent gasUsed is the same as the target, the baseFee remains unchanged.
-		if parent.GasUsed == parentGasTarget {
-			return new(big.Int).Set(parent.BaseFee)
-		}
-		if parent.GasUsed > parentGasTarget {
-			// If the parent block used more gas than its target, the baseFee should increase.
-			gasUsedDelta := new(big.Int).SetUint64(parent.GasUsed - parentGasTarget)
-			x := new(big.Int).Mul(parent.BaseFee, gasUsedDelta)
-			y := x.Div(x, parentGasTargetBig)
-			baseFeeDelta := math.BigMax(
-				x.Div(y, baseFeeChangeDenominator),
-				common.Big1,
-			)
 
-			return x.Add(parent.BaseFee, baseFeeDelta)
-		} else {
-			// Otherwise if the parent block used less gas than its target, the baseFee should decrease.
-			gasUsedDelta := new(big.Int).SetUint64(parentGasTarget - parent.GasUsed)
-			x := new(big.Int).Mul(parent.BaseFee, gasUsedDelta)
-			y := x.Div(x, parentGasTargetBig)
-			baseFeeDelta := x.Div(y, baseFeeChangeDenominator)
-
-			return math.BigMax(
-				x.Sub(parent.BaseFee, baseFeeDelta),
-				common.Big0,
-			)
+		return x.Add(parent.BaseFee, baseFeeDelta)
+	} else {
+		// If the parent's base fee is at the minimum already, fast return the minimum base fee
+		minimumBaseFee := big.NewInt(params.InitialBaseFee)
```

Comment:
I think we should have another parameter for the minimum base fee instead of using the InitialBaseFee.

Reply:
fixed, PTAL